src/gluonts/dataset/parallelized_loader.py

```diff
@@ -346,6 +355,45 @@ def _worker_fn(
     return buf.getvalue()
 
 
+class _ShuffleBatchIter(Iterator[DataEntry]):
+    def __init__(
+        self,
+        base_iterator: Iterator[DataEntry],
+        shuffle_buffer_length: Optional[int],
```

Comment:
I would not make this optional: if one use this class, it’s because they want to shuffle.

The decision on whether to wrap (hence shuffle) the base_iterator or not (ie using it as it is) should be taken by the caller.

Reply:
I've changed this.